src/core/BigDecimal.ts

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A wrapper for BigDecimal values. You can use this class to store and query BigDecimal values in distributed objects.
+ *
+ * A `BigDecimal` consists of an arbitrary precision number {@link unscaledValue} and a {@link scale}. If zero or positive,
```

Comment:
```suggestion
 * A BigDecimal consists of an arbitrary precision number {@link unscaledValue} and a {@link scale}. If zero or positive,
```
Or put the other places we say `BigDecimal` into backticks